scripts/install-hooks.sh

```diff
@@ -8,7 +8,7 @@ GIT_DIR=$(git rev-parse --git-dir)
 echo "Installing hooks..."
 cp ./scripts/hook.sh $GIT_DIR/hooks/pre-commit
 chmod +x $GIT_DIR/hooks/pre-commit
-sed -i 's/commit/push/g' ./scripts/hook.sh
+sed -i '' 's/commit/push/g' scripts/hook.sh
 cp ./scripts/hook.sh $GIT_DIR/hooks/pre-push
```

Comment:
I see what is happening here; with the `sed` command we also change the executed tests from `tests/test_commit.py` to `tests/test_push.py`.  This is not a good way to run the command, however, because this code has side-effects: it modifies script/hook.sh.

Instead, could you create a new file, `template_hook.sh` with some temporary variable `<stage>` that you replace with `sed`? So you would run a `sed` command for each the pre-commit hook and the pre-push hook, changing `<stage>` to `commit` or `push` respectively.

Ideally, this would be done via a variable to the template script, but `sed` is ok for now.

Also a nit: newline above the `sed` on line 11.

Also, what is the empty string `''` for in the `sed` command?

Reply:
I have added the `template_hook.sh` with some temporary variable `<stage>` and replacing it with the `commit/push` while installing the `push/commit hooks`.
`''`  in the `sed` command was to say that I am not taking the back up of the file before modifying. However, I have removed it from the command as I have found the command that is compatible with `MacOS and Linux` both.